utils/markdown/inlines.go

```diff
@@ -398,8 +398,8 @@ func (p *inlineParser) lookForLinkOrImage() {
 				p.inlines = append(p.inlines[:d.TextNode], inline)
 			} else {
 				p.inlines = append(p.inlines[:d.TextNode], inline)
-				for element := element.Prev(); element != nil; element = element.Prev() {
-					if d := element.Value.(*delimiter); d.Type == linkOpeningDelimiter {
+				for inlineElement := element.Prev(); inlineElement != nil; inlineElement = inlineElement.Prev() {
+					if d := inlineElement.Value.(*delimiter); d.Type == linkOpeningDelimiter {
```

Comment:
This code block is very suspicious. Please double check.

Reply:
I think is right.

Reply:
As per above, I don't think this falls under "suspicious", and agree with @jespino that this appears to be right. It might be nicer to break this up into more helper methods, but that seems out of scope.